src/mds/DamageTable.cc

```diff
@@ -175,11 +174,10 @@ bool DamageTable::notify_dirfrag(inodeno_t ino, frag_t frag,
     return true;
   }
 
-  auto key = DirFragIdent(ino, frag);
-  if (dirfrags.count(key) == 0) {
+  if (auto [it, inserted] = dirfrags.try_emplace(DirFragIdent(ino, frag)); inserted) {
```

Comment:
I understand that changes to notify_dentry are required since  dentries is a 2d map (std::map<DirFragIdent, std::map<DentryIdent, DamageEntryRef> > dentries ) and that should fix the tracker ticket.

Are the changes to notify_dirfrag and notify_remote_damaged really required to fix the tracker ticket?  dirfrags ( std::map<DirFragIdent, DamageEntryRef> dirfrags)  and remotes (std::map<inodeno_t, DamageEntryRef> remotes; ) are 1d maps.

Reply:
It's just to make the code consistent in style.